velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -59,6 +59,10 @@ void ReaderBase::loadFileMetaData() {
     stream = input_->read(
         fileLength_ - readSize, readSize, dwio::common::LogType::FOOTER);
   }
+  if (stream == nullptr) {
+    stream = input_->read(
+        fileLength_ - readSize, readSize, dwio::common::LogType::FOOTER);
+  }
```

Comment:
Hi @zhli1142015, do you think this change is reasonable? It is to avoid segment fault after https://github.com/oap-project/velox/pull/333.
The core dump issue can be reproduced with below newly added test.
```
TEST_F(ParquetTableScanTest, prune)
```

Reply:
Make sense to me, thanks.

Reply:
And also set `preloadFile_ ` as false in if block, thanks.

Reply:
Hello @rui-mo 
I debugged above test case locally, looks like we use `CachedBufferInput` instead of `BufferedInput`. I think this is the case my change did not cover. mabye we can change code as below, to avoid this issue:
`  preloadFile_ = (dynamic_cast<dwio::common::CachedBufferedInput*>(
                      input_.get()) == nullptr) &&
      (fileLength_ <= filePreloadThreshold_ ||
       fileLength_ <= directorySizeGuess_);
`

Reply:
Thank you! Updated.